bazel/cc_wrapper.py

```diff
@@ -29,18 +29,20 @@ def sanitize_flagfile(in_path, out_fd):
 
 
 # Is the arg a flag indicating that we're building for C++ (rather than C)?
-def is_cpp_flag(arg):
+def old_is_cpp_flag(arg):
   return arg in ["-static-libstdc++", "-stdlib=libc++", "-lstdc++", "-lc++"
                 ] or arg.startswith("-std=c++") or arg.startswith("-std=gnu++")
 
 
-def main():
-  # Append CXXFLAGS to correctly detect include paths for either libstdc++ or libc++.
-  if sys.argv[1:5] == ["-E", "-xc++", "-", "-v"]:
-    os.execv(envoy_real_cxx, [envoy_real_cxx] + sys.argv[1:] + shlex.split(envoy_cxxflags))
+# Is the arg a flag indicating that we're building for C++ (rather than C)?
+def new_is_cpp_flag(arg):
```

Comment:
Can't we strip whitespace in the flags before passing to (or inside) this function to avoid multiple source of truth?

Reply:
Done. Refactored both pairs of `old_`/`new_` functions into one function by abstracting away the argument processing to the `main` function.